### ISSUE TITLE
Issue with punctuation and context building

### ISSUE THREAD

**Issue by bunny-therapist** [quesurifn/yake-rust#14]
The created contexts contain punctuation symbols. If a word is just composed of punctuation symbols, it should be skipped and the buffer emptied.

I fixed this is my branch here: https://github.com/bunny-therapist/yake-rust/commit/a9c3a9917a49830ed5133b1ef6b1bcbe57b671e5

The relevant part in LIAAD/yake is here: https://github.com/LIAAD/yake/blob/master/yake/datarepresentation.py#L59
The "exclude" chars in LIAAD/yake are what is called "punctuation" in yake-rust.

**Comment by bunny-therapist** [comment]
@xamgore - just making sure you saw this. Maybe you found a simpler way to solve this?

**Comment by bunny-therapist** [comment]
Yeah, but vocabulary is different from candidates apparently. The "buffer/buffer_words" vec we have end up with elements like "!" and "?" which then affect ctx.0/1 and thus wr, and wl, and thus frequency and relatedness.

**Comment by xamgore** [comment]
Hm, right. Can the candidate words have punctuation inside like `abc!?def`?

**Comment by bunny-therapist** [comment]
Based on the python code, it skips words that are composed entirely of punctuation. So "abc!?def" would be ok, but "!?" would not.
